helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PerInstanceAccessor.java

```diff
@@ -308,20 +309,34 @@ public Response updateInstanceConfig(@PathParam("clusterId") String clusterId,
     }
     InstanceConfig instanceConfig = new InstanceConfig(record);
     ConfigAccessor configAccessor = getConfigAccessor();
+
     try {
+      /*
+       * Even if the instance is disabled, non-valid instance topology config will cause rebalance
+       * failure. We are doing the check whenever user updates InstanceConfig.
+       */
+      if (command == Command.delete || command == Command.update) {
+        validateDeltaTopologySettingInInstanceConfig(clusterId, instanceName, configAccessor,
```

Comment:
I suggest changing the method signature a bit: `boolean isDelete` -> `Command command`. You could put if..else check in this private method.
An nice article to help with clean such boolean parameter : https://medium.com/@amlcurran/clean-code-the-curse-of-a-boolean-parameter-c237a830b7a3